lib/gears/object.lua

```diff
@@ -122,9 +122,47 @@ function object:emit_signal(name, ...)
     end
 end
 
---- Returns a new object. You can call :emit_signal(), :disconnect_signal,
--- :connect_signal() and :add_signal() on the resulting object.
-local function new()
+local function get_miss(self, key)
+    local class = rawget(self, "_class")
+
+    if rawget(self, "get_"..key) then
+        return rawget(self, "get_"..key)(self)
+    elseif class and class["get_"..key] then
+        return class["get_"..key](self)
+    elseif class and class[key] then
+        return class[key]
+    else
+        return rawget(self, key)
+    end
+
+end
+
+local function set_miss(self, key, value)
+    local class = rawget(self, "_class")
+
+    if rawget(self, "set_"..key) then
+        return rawget(self, "set_"..key)(self, value)
+    elseif class and class["set_"..key] then
+        return class["set_"..key](self, value)
+    elseif rawget(self, "_enable_auto_signals") then
+        self:emit_signal("property::"..key, value)
+        class[key] = value
```

Comment:
You should set the new value before emitting the signal, no? (And, if you want, emit the signal with the old value as argument). You always cite Qt for this. :-)

Apropos Qt: Only do this if `new_value ~= old_value`.